src/main/java/org/zalando/nakadi/validation/EventTypeOptionsValidator.java

```diff
@@ -28,13 +30,16 @@ public void validate(final Object target, final Errors errors) {
     }
 
     private void checkRetentionTime(final Errors errors, final EventTypeOptions options) {
-        if (Objects.nonNull(options) && Objects.nonNull(options.getRetentionTime())) {
-            final Long retentionTime = options.getRetentionTime();
-            if (retentionTime > maxTopicRetentionMs) {
-                createError(errors, "can not be more than " + maxTopicRetentionMs);
-            } else if (retentionTime < minTopicRetentionMs) {
-                createError(errors, "can not be less than " + minTopicRetentionMs);
-            }
+        final Long retentionTime = options.getRetentionTime();
+        if (retentionTime == null) {
+            options.setRetentionTime(defaultTopicRetentionMs);
```

Comment:
I agree that it looks a bit strange but I think it is better place to set the default value otherwise we have to do it somewhere in controller or service where we will have to inject KafkaRepositorySettings to get the default value for retention time which is not very suitable as well. Also I tried approach with json deserialization which didn't work our for me. What is your point?